File: gameSrc/model/classes/property.js
//
// @class
// @extends cwt.IndexMultiton
//
cwt.PropertyClass = my.Class({

  STATIC: {

    fromJSON: function(data) {

    },

    toJSON: function() {

    },

    CAPTURE_POINTS: 20,

    CAPTURE_STEP: 10
  },

  constructor: function() {
    this.points = 20;

    //
    // @type {cwt.Player}
    //
    this.owner = null;

    this.type = null;
  },

  //
  // Returns true, when the given property is neutral, else false.
  //
  isNeutral: function() {
    return this.owner === null;
  },

  makeNeutral: function() {
    this.owner = null;
  }

});

File: gameSrc/org/wolftec/cwt/model/Property.java
package org.wolftec.cwt.model;

import org.wolftec.cwt.sheets.PropertyType;

public class Property {

  public static final int CAPTURE_POINTS = 20; // TODO
  public static final int CAPTURE_STEP   = 10; // TODO

  public int              points         = 20;
  public Player           owner;
  public PropertyType     type;

  public Property() {
    this.points = 20;
    this.owner = null;
    this.type = null;
  }

  /**
   * @return true, when the given property is neutral, else false.
   */
  public boolean isNeutral() {
    return owner == null;
  }

  public void makeNeutral() {
    owner = null;
  }
}
